synthesis/codon/analyzer.go

```diff
@@ -0,0 +1,36 @@
+package codon
+
+import "github.com/TimothyStiles/poly/io/genbank"
+
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		StartCodonCount: map[string]int{},
+	}
+}
+
+// Analyzer is used to compute codon usage statistics
+type Analyzer struct{}
+
+// ComputeStats returns a set of codon related statistics from a genebank sequence
+func (a *Analyzer) ComputeStats(sequence genbank.Genbank) (*Stats, error) {
+	stats := NewStats()
+
+	// iterate through the features of the genbank file and if the feature is a coding region, get the start codon
+	for _, feature := range sequence.Features {
+		if feature.Type == "CDS" {
```

Comment:
@Koeng101 hmm, maybe we should store common feature types in the genbank package as an enum (`type FeatureType string`).

Reply:
That's a good idea! Though first we have to merge the genbank package... maybe add as an issue?